pkg/bootstrap/docker/openshift/helper.go

```diff
@@ -273,48 +287,63 @@ func (h *Helper) Start(opt *StartOptions, out io.Writer) (string, error) {
 	}
 	skipCreateConfig := false
 	if opt.UseExistingConfig {
+		masterConfigExists := false
+		nodeConfigExists := false
 		var err error
 		configDir, err = h.copyConfig()
 		if err == nil {
 			_, err = os.Stat(filepath.Join(configDir, "master", "master-config.yaml"))
 			if err == nil {
-				skipCreateConfig = true
+				masterConfigExists = true
+			}
+			_, err = os.Stat(filepath.Join(configDir, "node-localhost", "node-config.yaml"))
+			if err == nil {
+				nodeConfigExists = true
 			}
 		}
+		if masterConfigExists && nodeConfigExists {
+			skipCreateConfig = true
+		}
 	}
 
 	// Create configuration if needed
-	var nodeHost string
 	if !skipCreateConfig {
 		fmt.Fprintf(out, "Creating initial OpenShift configuration\n")
+		certHosts := append(defaultCertHosts, opt.AdditionalIPs...)
+		if len(h.publicHost) > 0 {
+			certHosts = append(certHosts, h.publicHost)
+		}
+		createCertsCmd := []string{
+			"admin", "ca", "create-master-certs",
+			"--cert-dir=/var/lib/origin/openshift.local.config/master",
+			"--master=127.0.0.1",
+			fmt.Sprintf("--hostnames=%s", strings.Join(certHosts, ",")),
+		}
+		if len(h.publicHost) > 0 {
+			createCertsCmd = append(createCertsCmd, fmt.Sprintf("--public-master=https://%s:8443", h.publicHost))
+		}
+		_, err := h.runHelper.New().Image(h.image).
+			Privileged().
+			DiscardContainer().
+			HostNetwork().
+			HostPid().
+			Bind(binds...).
+			Env(env...).
+			Command(createCertsCmd...).Run()
+
 		createConfigCmd := []string{
 			"start",
 			fmt.Sprintf("--images=%s", opt.Images),
 			fmt.Sprintf("--volume-dir=%s", opt.HostVolumesDir),
 			fmt.Sprintf("--dns=0.0.0.0:%d", opt.DNSPort),
 			"--write-config=/var/lib/origin/openshift.local.config",
+			"--master=127.0.0.1",
+			"--hostname=localhost",
```

Comment:
It all depends on what's in /etc/hosts and how nsswitch is set up, and whether "nss-myhostname" is installed as well.

But:

```
$ getent hosts localhost
::1             localhost
$ getent ahosts localhost
127.0.0.1       STREAM localhost.localdomain
127.0.0.1       DGRAM  
127.0.0.1       RAW    
```
'getent hosts' uses gethostbyname2(), while 'ahosts' uses getaddrinfo() so depending on how the program actually resolves localhost, yeah it might well return ::1.